src/wp-admin/includes/image.php

```diff
@@ -278,70 +282,98 @@ function wp_create_image_subsizes( $file, $attachment_id ) {
 		 */
 		$threshold = (int) apply_filters( 'big_image_size_threshold', 2560, $imagesize, $file, $attachment_id );
 
-		// If the original image's dimensions are over the threshold,
-		// scale the image and use it as the "full" size.
-		if ( $threshold && ( $image_meta['width'] > $threshold || $image_meta['height'] > $threshold ) ) {
+		// Generate full sized images for all mime types.
+		foreach ( $output_mime_types as $mime_index => $output_mime_type ) {
```

Comment:
yea, makes sense and I had written it this way originally and decided to combine the code into a single loop in https://github.com/WordPress/wordpress-develop/pull/2393#:~:text=%E2%80%A6-,f2d0893,-remove%20duplicate%20editor otherwise i was wound up writing a bunch of the loop code twice. 

I see what you mean about being easier to parse though so I'll try restructuring as you suggest in a single commit so we can review.

Reply:
refactored into existing structure